### DISCUSSION TITLE
ClassData that implements IDisposable

### DISCUSSION THREAD

**Discussion by jgowdy** [xunit/xunit#2160] [Question] [answered]
It seems like ClassData classes that implement IDisposable aren't disposed.  Is this expected behavior?

**Comment by jgowdy** [comment, marked as answer]
Created this PR to attempt to fix the issue: https://github.com/xunit/xunit/pull/2161

**Reply by bradwilson** [reply]
Thanks, will take a look at the PR and take it up there! Also worth noting that this code (not just yours, but the data disposal in general) doesn't yet seem to support `IAsyncDisposable`.
